drivers/crypto/crypto_stm32_hash_priv.h

```diff
@@ -37,4 +46,40 @@ struct crypto_stm32_hash_session {
 #define CRYPTO_STM32_HASH_SESSN(ctx)                                                               \
 	((struct crypto_stm32_hash_session *const)(ctx)->drv_sessn_state)
 
+
+static inline HAL_StatusTypeDef hal_func_hash_SHA224_start(HASH_HandleTypeDef *hhash,
+					void *p_in_buffer,
+					uint32_t in_size_byte,
+					void *p_out_buffer)
+{
+#if defined(CONFIG_SOC_SERIES_STM32H7RSX)
+	return HAL_HASH_Start(hhash,
+			(const uint8_t *const)p_in_buffer,
+#else
+	return HAL_HASHEx_SHA224_Start(hhash,
+			p_in_buffer,
+#endif /* CONFIG_SOC_SERIES_STM32H7RSX */
+			in_size_byte,
+			p_out_buffer,
+			HAL_MAX_DELAY);
+}
+
+static inline HAL_StatusTypeDef hal_func_hash_SHA256_start(HASH_HandleTypeDef *hhash,
+					void *p_in_buffer,
+					uint32_t in_size_byte,
+					void *p_out_buffer)
+{
+#if defined(CONFIG_SOC_SERIES_STM32H7RSX)
+	return HAL_HASH_Start(hhash,
+			(const uint8_t *const)p_in_buffer,
+#else
+	return HAL_HASHEx_SHA256_Start(hhash,
+			p_in_buffer,
+#endif /* CONFIG_SOC_SERIES_STM32H7RSX */
```

Comment:
```suggestion
#if defined(CONFIG_SOC_SERIES_STM32H7RSX)
	return HAL_HASH_Start(hhash, (const uint8_t *const)p_in_buffer,
#else
	return HAL_HASHEx_SHA256_Start(hhash, p_in_buffer,
#endif /* CONFIG_SOC_SERIES_STM32H7RSX */
```

Reply:
I would rather be in favor of a more readable way:
```c
#if defined(CONFIG_SOC_SERIES_STM32H7RSX)
	return HAL_HASH_Start(hhash, (const uint8_t *const)p_in_buffer, in_size_byte, p_out_buffer,
			      HAL_MAX_DELAY);
#else
	return HAL_HASHEx_SHA256_Start(hhash, p_in_buffer, in_size_byte, p_out_buffer,
				       HAL_MAX_DELAY);
#endif /* CONFIG_SOC_SERIES_STM32H7RSX */
```

Ditto in `hal_func_hash_SHA224_start()`.